wrappers/services.go

```diff
@@ -471,13 +468,17 @@ func tryFileUpdate(path string, desiredContent []byte) (old *osutil.MemoryFileSt
 	}
 }
 
-type SnapServiceOptions struct {
+type SnapServicesOptions struct {
 	// VitalityRank is the rank of all services in the specified snap used by
 	// the OOM killer when OOM conditions are reached.
 	VitalityRank int
 
-	// QuotaGroup is the quota group for all services in the specified snap.
+	// QuotaGroup is the quota group for the specified snap.
 	QuotaGroup *quota.Group
+
+	// ServiceQuotaMap is the map of snap services that have been affected and should
+	// have their services updated.
+	ServiceQuotaMap map[*snap.AppInfo]*quota.Group
```

Comment:
I still have the impression that defined this way and for robustness, we should support having nil entries in this that then map back to QuotaGroup as we need that one anyway

Reply:
Fixed! I moved this handling of nil entries down to EnsureSnapServices